File: octo/mns/mns-cache/src/main/scala/com/sankuai/octo/mnsc/model/Env.scala
package com.sankuai.octo.mnsc.model

import com.sankuai.inf.octo.mns.util.ProcessInfoUtil
import org.apache.commons.lang.StringUtils

object Env extends Enumeration {
  type Env = Value
  val test = Value(1)
  val stage = Value(2)
  val prod = Value(3)

  def isValid(env: String) = {
    if (StringUtils.isEmpty(env)) {
      false
    } else {
      if (ProcessInfoUtil.isLocalHostOnline) {
        onlineEnv.contains(env)
      } else {
        offlineEnv.contains(env)
      }
    }
  }

  def strConvertEnum(env:String)={
   val envEnum= if (ProcessInfoUtil.isLocalHostOnline){
      onlineEnv.get(env)
    }else{
      offlineEnv.get(env)
    }
    envEnum.get
  }

  private val offlineEnv = Map(
    "ppe" -> stage,
    "dev" -> prod,
    "test" -> test,
    "prod" -> prod,
    "stage" -> stage,
    "beta" -> stage,
    "1" -> test,
    "2" -> stage,
    "3" -> prod
  )
  private val onlineEnv = Map(
    "prod" -> prod,
    "staging" -> stage,
    "stage" -> stage,
    "test" -> test,
    "1" -> test,
    "2" -> stage,
    "3" -> prod
  )

  def isValid(env: Int) = values.map(_.id).contains(env)
}

File: octo/mcc/java/config-server/src/main/java/com/sankuai/meituan/config/model/Env.java
package com.sankuai.meituan.config.model;


import com.sankuai.meituan.config.util.Common;
import org.apache.commons.lang3.StringUtils;

/**
 * Created by lhmily on 06/12/2016.
 */
public enum Env {
    test(1), stage(2), prod(3);
    public int id;

    Env(int i) {
        this.id = i;
    }

    @Override
    public String toString() {
        return this.name();
    }

    public static Env get(int i) {
        if (i < 1 || i > 3) throw new IllegalArgumentException("Invalid env id = " + i);
        return 3 == i ? prod : (2 == i ? stage : test);
    }

    public static String correctShowEnv(String oldEnv) {
        String ret = oldEnv;
        if (Common.isOnline()) {
            if ("stage".equals(oldEnv)) {
                ret = "staging";
            }
        } else {
            if ("stage".equals(oldEnv)) {
                ret = "beta";
            } else if ("prod".equals(oldEnv)) {
                ret = "dev";
            }
        }
        return ret;
    }
    public static boolean isValid(String env){
        return StringUtils.equals("test", env) || StringUtils.equals("stage", env) || StringUtils.equals("prod", env);
    }
}
